docs/documentation/features/frameworkUtilities.md

```diff
@@ -6,9 +6,116 @@ layout: documentation
 
 In this chapter useful services/utilities of the Eclipse SmartHome project are described. 
 
+## Network Server TLS service
+
+This service is responsible for providing SSLContexts that can be used all over ESH
+for server applications like webservlets, Mqtt servers or any extension that need to
+host a service. It can provide a SSLContext for TCP/TLS usage,
+as well as Connector for UDP/DTLS use cases.
+
+The SSLContext is configured to use either the user provided java keystore file
+or if none is provided to generate a self-signed certificate and public/private key-pair.
+
+The service can be asked to return different SSLContext or Connector objects based on a given
+context string. The default context will be returned if null is passed as context or if the
+given context is not configured.
+
+If nothing is configured (first-start), the service will generate a self-signed certificate and public/private key-pair in `ESH_DIR/etc/default.keystore` and store a new configuration file in `ESH_DIR/etc/networkServerTlsProvider.cfg`, including the keystore access password and private key password.
+
+You can provide your own java keystore files in `ESH_DIR/etc/yourKeystoreFile.keystore`. The name of the file is automatically the context name, e.g. you would request a `SSLContext` like this: `createSSLContext("yourKeystoreFile")`.
+Don't forget to add the configuration for your context to the configuration file as explained in the next section.
+
+The service can use ACME to automatically sign a public key by the Let's Encrypt initiative. 
+
+### Configuration
+
+The service accepts a configuration file `networkServerTlsProvider.cfg` with the following parameters:
+
+* **domains**: for instance "www.abc.org|default,www.def.org|second". *¹ *²
+* **psks**: for instance "JVBER|default,i0xLjUK|second". *¹ *²
+* **keystorePasswords**: for instance "JVBER|default,i0xLjUK|second". *¹ *²
```

Comment:
I guess encoding/decoding passwords to Base64 is done basically because they can be concatenated this way. This would also go away when the configuration structure would change.

Reply:
Nope it has to do with the fact that passwords don't need to be ASCII but can be any characters, even those that break the file for editors.